### ISSUE TITLE
403 with airflow 1.10 version

### ISSUE THREAD

**Issue by guptashailesh92** [teamclairvoyant/airflow-rest-api-plugin#41]
I tried it with airflow 1.10 and wasnt able to make it work. It was throwing 403, forbidden. I downgraded the version to 1.8 then it was working fine. Did anyone else faced this issue? Any workaround?

**Comment by sn95racing** [comment]
I also have the same problem. Note that I am using the puckel/airflow docker image (hopefully this is the right place to post it). I tried with airflow version 1.10.1 and 1.10.3 and received a 403 error when trying to use the api. 

**This is the configuration I have in airflow.config:**
[rest_api_plugin]

log_loading=True
filter_loading_messages_in_cli_response=False
rest_api_plugin_http_token_header_name=user
rest_api_plugin_expected_http_token=pass

**I get this response inside the webserver logs:**

Initialization:
webserver_1  | [2019-07-09 18:24:19,524] {{__init__.py:51}} INFO - Using executor CeleryExecutor

webserver_1  | [2019-07-09 18:24:19,615] {{rest_api_plugin.py:48}} INFO - Initializing Airflow REST API 

Plugin with configs:
webserver_1  | [2019-07-09 18:24:19,616] {{rest_api_plugin.py:49}} INFO -       rest_api_endpoint: /admin/rest_api/api

webserver_1  | [2019-07-09 18:24:19,616] {{rest_api_plugin.py:50}} INFO -       hostname: 4fbf369359a9

webserver_1  | [2019-07-09 18:24:19,616] {{rest_api_plugin.py:51}} INFO -       airflow_version: 1.10.3

webserver_1  | [2019-07-09 18:24:19,617] {{rest_api_plugin.py:52}} INFO -       rest_api_plugin_version: 1.0.6

webserver_1  | [2019-07-09 18:24:19,617] {{rest_api_plugin.py:53}} INFO -       airflow_webserver_base_url: http://localhost:8080

webserver_1  | [2019-07-09 18:24:19,617] {{rest_api_plugin.py:54}} INFO -       airflow_base_log_folder: /usr/local/airflow/logs

webserver_1  | [2019-07-09 18:24:19,618] {{rest_api_plugin.py:55}} INFO -       airflow_dags_folder: /usr/local/airflow/dags

webserver_1  | [2019-07-09 18:24:19,618] {{rest_api_plugin.py:56}} INFO -       airflow_rest_api_plugin_http_token_header_name: user

webserver_1  | [2019-07-09 18:24:19,618] {{rest_api_plugin.py:57}} INFO -       airflow_expected_http_token: OMITTED_FOR_SECURITY

webserver_1  | [2019-07-09 18:24:19,619] {{rest_api_plugin.py:58}} INFO -       filter_loading_messages_in_cli_response: False

**Making GET request:** 

`curl --header "user:pass"  http://localhost:8080/admin/rest_api/api?rest_api_plugin_version
`

**Response from webserver:**

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>403 Forbidden</title>
<h1>Forbidden</h1>
<p>You don't have the permission to access the requested resource. It is either read-protected or not readable by the server.</p>

**Webserver log after request:** 

`webserver_1  | 192.168.112.1 - - [09/Jul/2019:18:25:33 +0000] "GET /admin/rest_api/api?rest_api_plugin_version HTTP/1.1" 403 234 "-" "curl/7.57.0"
`

I can provide the full log and configuration files if necessary.

**Comment by rssanders3** [comment]
Hello @guptashailesh92 and @sn95racing, this is due to a recent fix that we made to the codebase for security purposes to not allow users to access the REST API from the Web server unless they're authenticated. We're working on a more permanent fix (which will eventually be in the 1.0.7 release), but for now, to get past this issue, you can **comment out** the function at line number 520 in the rest_api_plugin.py file (https://github.com/teamclairvoyant/airflow-rest-api-plugin/blob/v1.0.7-branch/plugins/rest_api_plugin.py#L520). This function:
```
    # overrides BaseView method to check permission for the menu link
    def is_accessible(self):
        return current_user.is_authenticated
```

**Comment by sudhakar-nallam** [comment]
This issue is resolved in #48
